canonicalwebteam/discourse/parsers/docs.py

```diff
@@ -709,3 +712,35 @@ def _replace_tutorials(self, tutorial_tables, tutorial_data):
             table["soup_table"].replace_with(
                 BeautifulSoup(card, features="html.parser")
             )
+
+    def _generate_headings_map(self, soup):
+        headings_map = []
+        current_list = headings_map
+        previous_tag = None
+
+        headings = soup.findAll(re.compile("^h[2-3]$"))
```

Comment:
The point of having a real parser like beautiful soup is specifically so you don't have to use regex to find stuff. It's dangerous to try to use Regex to parse HTML. See https://robinwinslow.uk/regex#avoid-coding-in-regex-if-you-can #shamelessSelfPromotion

Reply:
Sorry misunderstood. This is fine.

Reply:
I have updated it to use an array with the target headings `["h2","h3"]`, as this is more readable